tests/templates/test_templates.py

```diff
@@ -550,6 +550,55 @@ def test_get_type_by_name(self, client, mock_template):
         new_type = client.get_templatetype_by_name(type.template_id, type.name)
         assert new_type is not None, "Resource type attrs not retrived by name!"
 
+    def test_clone_templatetype(self, client, mock_template):
+        """Test cloning a template type"""
+        original_template = client.get_template(mock_template.id)
+        original_type = original_template.templatetypes[0]
+        
+        # Clone the template type
+        cloned_type = client.clone_templatetype(original_type.id)
+        cloned_type_j = JSONObject(cloned_type)
+        
+        # Verify the clone was created successfully
+        assert cloned_type_j is not None, "Cloned template type should not be None"
+        assert cloned_type_j.id is not None, "Cloned template type should have an ID"
+        assert cloned_type_j.id != original_type.id, "Cloned type should have different ID than original"
+        
+        # Verify that only the specified fields are different
+        # Name should be modified to be unique (appends "(Clone)")
+        assert cloned_type_j.name == original_type.name + " (Clone)", "Cloned type should have modified name"
+        assert cloned_type_j.alias == original_type.alias, "Cloned type should have same alias as original"
+        assert cloned_type_j.description == original_type.description, "Cloned type should have same description as original"
+        assert cloned_type_j.resource_type == original_type.resource_type, "Cloned type should have same resource_type as original"
+        assert cloned_type_j.template_id == original_type.template_id, "Cloned type should have same template_id as original"
+        assert cloned_type_j.layout == original_type.layout, "Cloned type should have same layout as original"
+        
+        # Verify that cr_date could be different (but might be same if created in same second)
+        # The important thing is that the clone has a valid cr_date
+        assert cloned_type_j.cr_date is not None, "Cloned type should have a creation date"
```

Comment:
This assertion is failing in the test for me; the other attrs tested for are present and agree with the original, but there is no `cr_date` on the `JSONObject`.

Reply:
removed